lwt/xen_api_lwt_unix.ml

```diff
@@ -62,12 +62,17 @@ module Lwt_unix_IO = struct
     Ssl.create_context Ssl.TLSv1_2 Ssl.Client_context
 
   let open_connection uri =
-    let domain_addr_t = match Uri.host uri with
+    let domain_addr_t port = match Uri.host uri with
       | Some host ->
         Lwt.catch
           (fun () ->
-             Lwt_unix.gethostbyname host >>= fun host_entry ->
-             return (host_entry.Lwt_unix.h_addrtype, host_entry.Lwt_unix.h_addr_list.(0)))
+             let host = Scanf.ksscanf host (fun _ _ -> host) "[%s@]" Fun.id in
```

Comment:
These bits of code that you have changed seem to do similar things. Can we factor out some of it? For example (i didn't try to compile):

```
let sockaddr_of_uri uri port =
  match Uri.host uri with
  | None -> fail (Failed_to_resolve_hostname "")
  | Some host ->
      Lwt.catch (fun () ->
        let host = Scanf.ksscanf host (fun _ _ -> host) "[%s@]" Fun.id in
        Lwt_unix.getaddrinfo host (string_of_int port) [] >|= function
          | [] -> None
          | addrinfo :: _ -> Some addrinfo.Unix.ai_addr)
        (fun () ->  fail (Failed_to_resolve_hostname host))
```

Reply:
Yes indeed, where should I put this utility method?

Reply:
Maybe in a util.ml file in this folder, it's up to you really